### PR TITLE
🎨 App Layout - Create Website Footer

### LINKED ISSUE #135: 🎨 App Layout - Create Website Footer
#  Create the website footer
---
Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)
---

### Summary

As a user I want a simple footer that displays my information in a structured way.
Another component can be injected into the footer to highlight a certain action per page

### Description

Currently the website page has no footer. Your job is to add one. The footer needs to appear well on web and mobile:

Web:

dynamic footer

Mobile:

dynamic footer mobile

NB: The mobile design is for a different website, ours should look exactly the same apart from the textual content itself

---
Link to full design:

```
https://xd.adobe.com/view/30f592fb-2ceb-4ea9-a6b9-2aee6239a957-65a6/screen/9ce6f3d4-3ff5-4c36-86f0-bf3a67e02b13
```

### Technical Analysis

Create a layout package in libs/elements/layout/footer

Create a footer component inside a components folder of the package above.
Add a module and export the component.

The footer has a dynamic component which can change from page to page, as illustrated below:

dynamic footer transclution

Use simple content projection/transclusion for this through the `<ng-content></ng-content>` element

https://angular.io/guide/content-projection

#### Social media links

https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690295/elewa-group-website/Icons/SVG/_Social/instagram_x8ujcq.svg

https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690294/elewa-group-website/Icons/SVG/_Social/linkedin_x8smhm.svg

https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690294/elewa-group-website/Icons/SVG/_Social/facebook_lhhwk2.svg

### Acceptance Criteria

- [ ] When rendered the component is pixel perfect compared to the design above.
- [ ] The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
- [ ] The code is isolated to the libs/elements/layout/footer package, unless a justified reason is specified.
- [ ] The design is responsive for screen, tablet and phone
- [ ] The component allows for a transclusive element that is centered using CSS

### PR DESCRIPTION
**Pull request by Paul-ike**
# Description

Created a Website Footer component <elewa-website-footer>, this component can be reused anywhere in the site.

Fixes #135 🎨 App Layout - Create Website Footer

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshot (optional)
Screenshot
Screenshot

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Beulah-Matt** [comment]
@Paul-ike Could you add screenshots of your project?

File: libs/data/ui/content-text/src/lib/mock-footer.data.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      id: "social-media-1",
+      icon: "https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690295/elewa-group-website/Icons/SVG/_Social/instagram_x8ujcq.svg",
+      link: "https://www.instagram.com/",
+    },
+    {
+      id: "social-media-4",
+      icon: "https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690294/elewa-group-website/Icons/SVG/_Social/linkedin_x8smhm.svg",
+      link: "https://www.linkedin.com/",
+    },
+  ];
```
Comment (Beulah-Matt) [inline libs/data/ui/content-text/src/lib/mock-footer.data.ts:61]: Good job with separation of concerns :+1:

File: libs/elements/layout/footer/src/lib/components/footer/footer.component.scss [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+.footer__links
+{
+    width: 100%;
+    display: flex;
+    justify-content: space-between;
+    flex-wrap: wrap;
+    margin-top: 10px;
+    border-top: 1px solid #aaaaaa;
```
Comment (Beulah-Matt) [inline libs/elements/layout/footer/src/lib/components/footer/footer.component.scss:14]: Use color variables from the variables.scss file. 
If they are not provided for, add them there.

**Comment by Paul-ike** [comment]
I have added

**Review by Beulah-Matt — CHANGES_REQUESTED** [review]
Generally a good job. Just make the few changes.

**Comment by Paul-ike** [comment]
I have commited the changes you requested.

**Pull request closed without merging**